libbeat/autodiscover/providers/kubernetes/node.go

```diff
@@ -0,0 +1,211 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
Interesting. This adds node auto discovery. 
While this might be interesting and useful to have in our 6.6 based version is it truly needed for supporting namespace labels?

Reply:
Yes, According to PR on Elastic/Beat regarding on this issue.
They have made this all required changes in each respected files.

I am following this:
https://github.com/elastic/beats/issues/13873
https://github.com/elastic/beats/pull/15528

Reply:
OK, then that is what we have to do.